### PR TITLE
chore: Migrate CI to GitHub Actions 

### PR DESCRIPTION
**Pull request by sawood14012**
# Description

Have added/updated scripts which enable git hub actions .

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cico_run_pylint.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ prep
 check_python_version
 ./qa/detect-common-errors.sh
 ./qa/detect-dead-code.sh
-./qa/measure-cyclomatic-complexity.sh
```
Comment (sawood14012) [inline cico_run_pylint.sh:22]: removed cc check in linter because unit tests script runs it

File: requirements.in [hunk 2]
```diff
@@ -7,12 +7,17 @@ uuid
 requests
 boto3
 boto
-psycopg2
+psycopg2==2.7.6.1
```
Comment (arajkumar) [inline requirements.in:10]: Use psycopg2-binary without version locking.

File: requirements.in [hunk 3]
```diff
@@ -7,12 +7,17 @@ uuid
...
 sqlalchemy
-minio
-gevent
+minio==4.0.8
```
Comment (arajkumar) [inline requirements.in:12]: Don't version lock on requirement.in unless otherwise needed

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: requirements.in (hunk 3 above)
Comment (sawood14012) [reply requirements.in:12]: i have only version locked if the newer versions cause problems

File: requirements.in (hunk 2 above)
Comment (sawood14012) [reply requirements.in:10]: okay

File: cico_run_pylint.sh [hunk 4]
```diff
@@ -19,7 +19,7 @@ prep
...
+#./qa/measure-cyclomatic-complexity.sh
 #./measure-cyclomatic-complexity.sh --fail-on-error
```
Comment (arajkumar) [inline cico_run_pylint.sh:23]: Don't comment, delete it if you don't need it.

File: requirements.in [hunk 5]
```diff
@@ -7,12 +7,17 @@ uuid
 requests
 boto3
 boto
-psycopg2
+psycopg2-binary
 sqlalchemy
-minio
-gevent
+minio==4.0.8
+mock
```
Comment (arajkumar) [inline requirements.in:13]: I see both normal and dev dependencies in the same file. Are you doing dependency cleanup as part of this work? In that case it has to be split into 2.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: cico_run_pylint.sh (hunk 4 above)
Comment (sawood14012) [reply cico_run_pylint.sh]: done

File: requirements.in (hunk 5 above)
Comment (sawood14012) [reply requirements.in:12]: done

File: requirements.in [hunk 6]
```diff
@@ -7,12 +6,13 @@ uuid
 requests
 boto3
 boto
-psycopg2
+psycopg2-binary
 sqlalchemy
-minio
-gevent
+minio==4.0.8
```
Comment (arajkumar) [inline requirements.in:11]: Isn't minio a test dependency?

File: requirements.in [hunk 7]
```diff
@@ -7,12 +6,13 @@ uuid
...
+mock
+gevent==1.4.0
 six
 raven[flask]
-pytest-mock
-pytest-flask
 radon
+werkzeug==0.16.0
```
Comment (arajkumar) [inline requirements.in:17]: Why this is version locked? None of our repos lock werkzeug.

File: qa/runtests.sh [hunk 8]
```diff
@@ -62,13 +62,8 @@ function setup_virtualenv {
...
     pip install -U pip
-    pip3.6 install -r requirements.txt
-
-    # Install profiling module
-    pip3.6 install pytest-profiling
-
-    # Install pytest-coverage module
-    pip3.6 install pytest-cov
+    pip install -r requirements.txt
+    pip install -r test/requirements.txt
```
Comment (arajkumar) [inline qa/runtests.sh:66]: Do a atomic install.
```suggestion
    pip install -r requirements.txt -r test/requirements.txt
```

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: requirements.in (hunk 7 above)
Comment (sawood14012) [reply requirements.in:17]: werkzeug has a newer versions eg: 1.0.0 but the package we use has been moved , if we dont version lock  werkzeug then we have to change some of the code 
ref :  `src/rest_api.py:388
  /Users/sawood/work/fabric8-analytics-data-model/src/rest_api.py:388: DeprecationWarning: 'werkzeug.contrib.fixers.ProxyFix' has moved to 'werkzeug.middleware.proxy_fix.ProxyFix'. This import is deprecated as of version 0.15 and will be removed in 1.0.
    app.wsgi_app = ProxyFix(app.wsgi_app)`

File: requirements.in (hunk 6 above)
Comment (sawood14012) [reply requirements.in:11]: yup

**Comment by sawood14012** [comment]
[test]

**Review by arajkumar — APPROVED** [review]
lgtm

**Comment by sawood14012** [comment]
[test]

**Pull request merged**